File: app/src/main/java/com/redteamobile/smart/cellular/TelephonySetting.java
package com.redteamobile.smart.cellular;

public interface TelephonySetting {

    String getImei();

    boolean getDataEnabled(int subId);

    void setDataEnabled(int subId, boolean enable);

    boolean getDataRoamingEnabled(int subId);

    void setDataRoamingEnabled(int subId, boolean enable);

    boolean setNonRoamingForOperator(String mccMnc);

    boolean setOperatorBrandName(String name);

    int getSubId(int slotId);

    String getImsi(int subId);

    String initIccid();

    boolean isSimReady(int slotId);

    int getDefaultDataSubId();

    void setDefaultDataSubId(int subId);

    void setDisplayName(int subId, String name);

    String getModel();

    String getNetworkType();

    boolean configureApn(String apn, String mccMnc);

    boolean clearApn();

}

File: app/src/main/java/com/redteamobile/smart/agent/cellular/TelephonySetting.kt
package com.redteamobile.smart.agent.cellular

interface TelephonySetting {

    fun getImei(): String

    // Privileged system permission required
    fun getDataEnabled(subId: Int): Boolean

    // Privileged system permission required
    fun setDataEnabled(subId: Int, enable: Boolean)

    fun getDataRoamingEnabled(subId: Int): Boolean

    // Privileged system permission required
    fun setDataRoamingEnabled(subId: Int, enable: Boolean)

    fun setNonRoamingForOperator(mccMnc: String): Boolean

    fun setOperatorBrandName(name: String): Boolean

    fun getSubId(slotId: Int): Int

    fun getImsi(subId: Int): String

    fun initIccid()

    fun isSimReady(slotId: Int): Boolean

    fun getDefaultDataSubId(): Int

    // Privileged system permission required
    fun setDefaultDataSubId(subId: Int)

    // Privileged system permission required
    fun setDisplayName(subId: Int, name: String)

    fun getModel(): String

    fun getNetworkType(): String

    fun configureApn(apn: String, mccMnc: String): Boolean

    fun clearApn(): Boolean
}